test/v1/controllers/DeleteLossClaimControllerSpec.scala

```diff
@@ -0,0 +1,128 @@
+
+    val controller = new DeleteLossClaimController(
+      authService = mockEnrolmentsAuthService,
+      lookupService = mockMtdIdLookupService,
+      deleteLossClaimService = mockDeleteLossClaimService,
+      deleteLossClaimParser = mockDeleteLossClaimRequestDataParser,
+      auditService = mockAuditService,
+      cc = cc
+    )
+
+    MockedMtdIdLookupService.lookup(nino).returns(Future.successful(Right("test-mtd-id")))
+    MockedEnrolmentsAuthService.authoriseUser()
+  }
+
+  "delete" should {
+    "return a successful response with header X-CorrelationId and body" when {
+      "the request received is valid" in new Test {
+
+        MockDeleteLossClaimRequestDataParser
+          .parseRequest(rawData)
+          .returns(Right(request))
+
+        MockDeleteLossClaimService
+          .delete(request)
+          .returns(Future.successful(Right(DesResponse(correlationId, ()))))
+
+        val result: Future[Result] = controller.delete(nino, claimId)(fakeRequest)
+        status(result) shouldBe NO_CONTENT
+      }
+    }
+
+    "handle mdtp validation errors as per spec" when {
+      def errorsFromParserTester(error: MtdError, expectedStatus: Int): Unit = {
+        s"a ${error.code} error is returned from the parser" in new Test {
+
+          MockDeleteLossClaimRequestDataParser
+            .parseRequest(rawData)
+            .returns(Left(ErrorWrapper(Some(correlationId), error, None)))
+
+          val response: Future[Result] = controller.delete(nino, claimId)(fakeRequest)
+
+          status(response) shouldBe expectedStatus
+          contentAsJson(response) shouldBe Json.toJson(error)
+          header("X-CorrelationId", response) shouldBe Some(correlationId)
+        }
+      }
+
+      errorsFromParserTester(BadRequestError, BAD_REQUEST)
+      errorsFromParserTester(NinoFormatError, BAD_REQUEST)
+      errorsFromParserTester(ClaimIdFormatError, BAD_REQUEST)
+
+    }
+
+    "handle non-mdtp validation errors as per spec" when {
+      def errorsFromServiceTester(error: MtdError, expectedStatus: Int): Unit = {
+        s"a ${error.code} error is returned from the service" in new Test {
+
+          MockDeleteLossClaimRequestDataParser
+            .parseRequest(rawData)
+            .returns(Right(request))
+
+          MockDeleteLossClaimService
+            .delete(request)
+            .returns(Future.successful(Left(ErrorWrapper(Some(correlationId), error, None))))
+
+          val response: Future[Result] = controller.delete(nino, claimId)(fakeRequest)
+          status(response) shouldBe expectedStatus
+          contentAsJson(response) shouldBe Json.toJson(error)
+          header("X-CorrelationId", response) shouldBe Some(correlationId)
+        }
+      }
+
+      errorsFromServiceTester(BadRequestError, BAD_REQUEST)
+      errorsFromServiceTester(NotFoundError, NOT_FOUND)
+      errorsFromServiceTester(NinoFormatError, BAD_REQUEST)
+      errorsFromServiceTester(ClaimIdFormatError, BAD_REQUEST)
```

Comment:
test for INTERNAL_SERVER_ERROR?

Reply:
OK. Good catch. Will add to this and all the others where we don't test for it.

Reply:
Done